veps/sig-network/hotpluggable-nad-ref.md

```diff
@@ -0,0 +1,191 @@
+This in turn calls various sub-logics that check for changes in areas like disk, volume, network, memory etc and decide if that requires a restart. The sub-logic for network
+would be updated to not require the `RestartRequired` condition if just the NAD reference (`networkName`) field has been changed.
+
+### Copying network specs from VM to VMI
+During the same Sync cycle, a logic deals checks if interfaces are being hotplugged or modified
+and copies the interfaces specs from VM to VMI specs accordingly. This would be adjusted to sync changes made to the `networkName` field as well.
+
+### Adding the correct migration conditions
+During the updateStatus cycle, the VMI controller in virt-controller evaluates the automatic migration request. Changes to interfaces and networks can trigger 3 states - immediate/pending/no migration.
+In case of hotplug/unplug of interfaces with supported bindings it requests either immediate migration condition (for SR-IOV interfaces) or pending migration (which is a delayed migration - 
+providing enough time for the `Dynamic Networks Controller`, if present to do an `InPlace` hotplug).
+This would be updated to request an immediate migration in case the interface connected to the changed network uses bridge binding.
+In case the VMI is live-migratable, as before, the WorkloadUpdateController will initiate a migration.
+
+#### Restricting update of source pod's network annotation
+In order to support in-place hotplug and hot unplug using the [Dynamic Networks Controller](https://github.com/k8snetworkplumbingwg/multus-dynamic-networks-controller) (DNC) the virt-controller updates the 
+multus annotation of the pod based on the difference between the VMI interface status and the specs. DNC then calculates the diff between the new and old annotation, attaches any new network and then removes deleted networks. 
+So trying to change the annotation from `'[{"name":"nad-with-vlan10","namespace":"default","mac":"02:87:f5:51:4c:5d","interface":"poda1363d52898"}]'` to `'[{"name":"nad-with-vlan20","namespace":"default","mac":"02:87:f5:51:4c:5d","interface":"poda1363d52898"}]'` will not work, as DNC will try to attach `nad-with-vlan20`
+without removing `nad-with-vlan10` causing a conflict in mac and interface id and thus failing the attachment.
+
+So, the reconciliation logic in virt-controller will be adjusted to not patch the annotation when only the NAD reference of an existing interface is updated.
+
+### Pros
+- Users can change the network of their VMs without a reboot.
+
+### Cons
+- In Place swapping of NAD referenced will not be possible even with Dynamic Networks Controller installed.
+
+## Alternative
+Instead of restricting update of source pod's network annotation, we can consider changing the logic in Dynamic Networks Controller to enable swapping networks (instead of adding and then removing).
+The rest of the design changes remain the same.
+
+### Pros
```

Comment:
Changing the NAD reference of non-migratable VMs when bridge binding is used - would be possible.